src/Windows.Core/DEVMODEW.cs

```diff
@@ -0,0 +1,783 @@
+﻿// Copyright © .NET Foundation and Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <summary>
+    /// Contains information about the initialization and environment of a printer or a display device.
+    /// </summary>
+    [StructLayout(LayoutKind.Explicit)]
+    public unsafe struct DEVMODEW
+    {
+        public const int CCHDEVICENAME = 32;
+        public const int CCHFORMNAME = 32;
+
+        /// <summary>
+        /// A zero-terminated character array that specifies the "friendly" name of the printer or display; for example, "PCL/HP LaserJet" in the case of PCL/HP LaserJet.
+        /// This string is unique among device drivers. Note that this name may be truncated to fit in the <see cref="dmDeviceName"/> array.
+        /// </summary>
+        [FieldOffset(0)]
+        public fixed char dmDeviceName[CCHDEVICENAME];
+
+        /// <summary>
+        /// The version number of the initialization data specification on which the structure is based.
+        /// To ensure the correct version is used for any operating system, use <c>DM_SPECVERSION</c>.
+        /// </summary>
+        [FieldOffset(32)]
```

Comment:
This needs to be 64 (and all subsequent fields adjusted accordingly). As it comes after a 32 `char` field, where `char` is 2 bytes.

Tip I learned from @qmfrederik: write a little VC++ program with output such as this:

```c
int main()
{
    std::cout << "dmSpecVersion offset: " << offsetof(DEVMODE, dmSpecVersion) << " size: " << sizeof(DEVMODE::dmSpecVersion) << "\n";
}
```

That prints out:

```
dmSpecVersion offset: 64 size: 2
```

Use this for any field you're not confident of the position or size.
Compile and run your program both for x86 and x64 to compare when in doubt about how pointer size may impact the layout.

Reply:
Lovely. I wasn't even aware of the `offsetof` operator in C++!
My C++ sucks anyway, otherwise I'd probably write my apps C++ in the first place...

Reply:
That's actually the output I got (don't worry, I haven't typed them all manually, I used copy paste, multiline-edit, and find-replace...):

```
dmDeviceName: 0
dmSpecVersion: 32       
dmDriverVersion: 34     
dmSize: 36
dmDriverExtra: 38       
dmFields: 40
dmOrientation: 44       
dmPaperSize: 46
dmPaperLength: 48
dmPaperWidth: 50
dmScale: 52
dmCopies: 54
dmDefaultSource: 56
dmPrintQuality: 58
dmPosition: 44
dmDisplayOrientation: 52
dmDisplayFixedOutput: 56
dmColor: 60
dmDuplex: 62
dmYResolution: 64
dmTTOption: 66
dmCollate: 68
dmFormName: 70
dmLogPixels: 102
dmBitsPerPel: 104
dmPelsWidth: 108
dmPelsHeight: 112
dmDisplayFlags: 116
dmNup: 116
dmDisplayFrequency: 120
dmICMMethod: 124
dmICMIntent: 128
dmMediaType: 132
dmDitherType: 136
dmReserved1: 140
dmReserved2: 144
dmPanningWidth: 148
dmPanningHeight: 152
```

Reply:
Last two offsets don't match with my commit. Will see if that fixes it.

Reply:
It didn't.

Reply:
OK running the `offsetof` with `DEVMODEW` gave me the desired values, and the test passes now.